tests/executors/test_executor_kill.py

```diff
@@ -37,15 +40,17 @@ def test_kill_custom_signal_kill():
 
 def test_already_closed():
     """Check that the executor cleans after itself after it exited earlier."""
-    with SimpleExecutor('python') as executor:
-        assert executor.running()
-        os.killpg(executor.process.pid, SIGKILL)
-
-        def process_stopped():
-            """Return True only only when self.process is not running."""
-            return executor.running() is False
-        executor.wait_for(process_stopped)
-        assert executor.process
+    with pytest.raises(ProcessExitedWithError) as excinfo:
```

Comment:
@karlb my only concern at the moment is that when using executor as context manager, in case of the process exiting with different exit code, you won't know whether it had issues when starting, or on exit.

As in the modified test here it will pass also when the process will have issues starting.

Reply:
Maybe simply subclassing the `ProcessExitedWithError` to `ProcessFinishedWithError` or something like that?

Reply:
Sounds good!